18 improved model getting and examples

MatInGit opened the pull request:
Addresses issue #18 
- Clean's up the model evaluation interface
- Adds examples for local and mlflow model
- Keeps supporting old models via legacy modules

Commit:
added lume-model examples, not working yet

Commit:
added intial docs

Commit:
untracked dir

Commit:
re-tracked

Commit:
sphinx doc worflow

Commit:
sphinx doc worflow

Commit:
sphinx doc worflow

Commit:
sphinx doc worflow

Commit:
sphinx doc workflow

Commit:
sphinx doc workflow

Commit:
sphinx doc workflow re-added static path

Commit:
sphinx doc workflow changed deploy script

Commit:
sphinx doc workflow changed deploy script

Commit:
sphinx doc workflow more deploy script changes

Commit:
sphinx doc workflow more deploy script changes

Commit:
sphinx doc workflow more deploy script changes

Commit:
changed makefile to make api docs

Commit:
changed makefile to make api docs

Commit:
changed makefile to make api docs

Commit:
changed makefile to make api docs

Commit:
changed the workflow to just use the built docs

Commit:
updated git ignore

Commit:
gitignore changes

Commit:
TOC change

Commit:
TOC change

Commit:
TOC change

Commit:
TOC change

Commit:
added warning

Commit:
added warning

Commit:
cool new theme

Commit:
clean docs build

Commit:
docs docs docs

Commit:
source

Commit:
all tests passing

Commit:
cleanup

Commit:
copy-pasta fix

Commit:
docs work

Commit:
more readable roadmap

Commit:
removed priority legend

Wikyboy commented:
In the case of the examples form examples/base/local, it would be good to have an initial setup for local users to run the examples through. While there is an explanation already in the readme.md on how to set up the environment it would be useful to have it repeat it in the example section for a new user.

Wikyboy commented:
This is more of a suggestion but I think some of these additions might be useful in a future release:
- A local image and compose to create a container for mlflow to allow for runing integration tests and allowing the examples to be run out of the box.
- An open source lume image on the DockerHub for easier deployment (this is based on a talk with the main contributer).
- Perhaps changing the image name on this open source repo from harbor to something else. It could be made based on a variable passed to the build push script. Changing the build and push script as well to reflect this more generic approach to allow for full customization for different users.
- Instructions for contributors.

Commit:
finished readme for release

MatInGit committed:
Merge pull request #22 from MatInGit/20-documentation

Improved docs

Commit:
added branch 18 to ci/cd for testing

Commit:
pypi publish

Commit:
fixed publish action

Commit:
skipping test if mlflow not reachable

Commit:
fixed testing

Commit:
removed workspace file

Commit:
test change

Commit:
fixed test pypi repo

MatInGit committed:
Added support for arrays and fixed handling in the fudge fix

MatInGit committed:
Refactor SimplePVAInterfaceServer initialization and variable handling for array types

MatInGit committed:
Add script to build and push Docker image to harbor.stfc.ac.uk

MatInGit committed:
Refactor SimplePVAInterfaceServer initialization and variable handling for array types. add port config to EPICS

MatInGit committed:
Re-enabled tests

MatInGit committed:
less annoying debug

MatInGit committed:
Arary changes finaliesed, needs a  test still

MatInGit committed:
Ruff format and fix

MatInGit committed:
chore: Update .gitignore and add Dockerfile.interactive

MatInGit committed:
fixed else if statements to catch the correct type

MatInGit committed:
ruff linter pass

MatInGit committed:
added test and implemented defaults for scalars and arrays

MatInGit committed:
tests for array inputs to sympy

MatInGit committed:
tests for array inputs to sympy

MatInGit committed:
commendted out code that didnt make sense

MatInGit committed:
cleanup and pyrpoject setup

MatInGit committed:
added worflow

MatInGit committed:
adjusted python versions

MatInGit committed:
changed workflow and removed python version reqirement

MatInGit committed:
changed workflow and removed python version reqirement

MatInGit committed:
changed workflow and removed python version reqirement

MatInGit committed:
added pythonpath to tests

MatInGit committed:
forgot install step after build

MatInGit committed:
changed testing setup to use hatch

MatInGit committed:
changed testing setup to use hatch

MatInGit committed:
re-enabled tests

MatInGit committed:
added versions reqs

MatInGit committed:
syntax in workflow

MatInGit committed:
dropped argparse from deps

MatInGit committed:
changed relative imports to module ones

MatInGit committed:
re-added deps and added minimum versions rather than strict ones

MatInGit committed:
changed _version.py dir

MatInGit committed:
refactor dependency management to use minimum version specifications

MatInGit committed:
removed argparse

MatInGit committed:
added pre-install to handle lume-model dep

MatInGit committed:
small changes

MatInGit committed:
deps deps deps

MatInGit committed:
attempt git workflow again

MatInGit committed:
change relative to absolute module pathing

MatInGit committed:
attempt git workflow again

MatInGit committed:
workflow change

MatInGit committed:
Final workflow cleanup

MatInGit committed:
workflow change

MatInGit committed:
workflow change

MatInGit committed:
removed unsupported versions

MatInGit committed:
fixed fail fast strategy

MatInGit committed:
dockerfile updates

MatInGit committed:
commented out useless print statements

MatInGit committed:
added 3.12 and changed workflow sligtly

MatInGit committed:
cicd

MatInGit committed:
added 3.12 and changed workflow sligtly

MatInGit committed:
added a base class for transformer to help implement new interfaces

MatInGit committed:
cicd undo

MatInGit committed:
added typing to abstract class

MatInGit committed:
tidy up

MatInGit committed:
Merge pull request #15 from MatInGit/transform-abstract-classes

[MERGE] Transform abstract classes added

MatInGit committed:
added dist to git ignore

MatInGit committed:
ruff format and fix, test passed

MatInGit committed:
tried adding profilinf to test but not very useful at this stage

MatInGit committed:
cleaned up unnecessary files

MatInGit committed:
made all transformer inherit from base class

MatInGit committed:
changed pydatic to v2

MatInGit committed:
added messaging intial protos

MatInGit committed:
changed test python exec to use sys.executable

MatInGit committed:
inetrface observer and test

MatInGit committed:
basic working chaining

MatInGit committed:
inetrface observer and test

MatInGit committed:
chaining test

MatInGit committed:
all test passing

MatInGit committed:
all test passing

MatInGit committed:
Merge branch '17-refactor-internal-message-passing' of https://github.com/MatInGit/lume-deployment into 17-refactor-internal-message-passing

MatInGit committed:
spelling

MatInGit committed:
commit before a simplification of the config system

MatInGit committed:
routing

MatInGit committed:
Refactor config parser and update test cases with new YAML schema

MatInGit committed:
Update subscription parameters in pv_mapping_isolated_nodes.yaml

MatInGit committed:
Enhance routing graph validation and update related tests

MatInGit committed:
black formatting all new tests

MatInGit committed:
black all other modified files

MatInGit committed:
all tests passing, bar some timeouts on occasion

MatInGit committed:
updated class and packages pngs

MatInGit committed:
removed old code

MatInGit committed:
removed old code

MatInGit committed:
removed commented out file for mlflowgetter

MatInGit committed:
Add optional rate parameter to DeploymentConfig, fix parse_queue typos, and update logging levels

MatInGit committed:
re-added one-shot run

MatInGit committed:
rationalising my tests

MatInGit committed:
started rewriting a readme

MatInGit committed:
fixed demo config

MatInGit committed:
readme

MatInGit committed:
removed the need to traverse outputs from transformers under tranformed keys

MatInGit committed:
made unpacking dictionaries default for modelObservers

MatInGit committed:
optimisations allowing high frequency refreshing

MatInGit committed:
some logging imrpovements

MatInGit committed:
sync

MatInGit committed:
add version flag to CLI for printing version information, clean up imports and remove old code

MatInGit committed:
fix: ensure newline at end of file in MlflowModelGetter.py

MatInGit committed:
chanign to poly-lithic naming

MatInGit committed:
rename: renamed model_manager to poly_lithic, removed lume_deploy naming to so it is consistent across the repo

MatInGit committed:
refactor: remove deprecated model_manager files

MatInGit committed:
cleanup files

MatInGit committed:
ruff formating

MatInGit committed:
refactor: update naming and improve version handling in CLI

MatInGit committed:
typo

MatInGit committed:
Merge branch '17-refactor-internal-message-passing' of github.com:MatInGit/lume-deployment into 17-refactor-internal-message-passing

MatInGit committed:
last changes before merge

MatInGit committed:
Merge pull request #19 from MatInGit/17-refactor-internal-message-passing

17 refactor internal message passing

MatInGit committed:
added gitlab build and test

MatInGit committed:
changed python version for lint

MatInGit committed:
changed python version for lint

MatInGit committed:
changed python version for lint

MatInGit committed:
changed python version for lint

MatInGit committed:
testfile

MatInGit committed:
testfile

MatInGit committed:
added pytorch to deps

MatInGit committed:
added pytorch to deps for now

MatInGit committed:
pytorch to torch

MatInGit committed:
copy-pasta package name

MatInGit committed:
copy-pasta package name

MatInGit committed:
reduced versions

MatInGit committed:
coverage regex copy-pasta

MatInGit committed:
reports

MatInGit committed:
stages

MatInGit committed:
report.xml

MatInGit committed:
added before script to coverage test

MatInGit committed:
made tests compatible lume-model 1.8.1

MatInGit committed:
added a quick uid comparison fix to ensure pvs are only updated on change

MatInGit committed:
fixed numpy version

MatInGit committed:
Refactor Dockerfile to create multi-stage builds for different environments and update build script to support new targets
Enhance messaging.py to use hashlib for message value hashing and improve observer notification timing statistics

MatInGit committed:
moved things around to clean up the repo, emptied out the examples dir

MatInGit committed:
push before new feature branch

MatInGit committed:
added intial docs

MatInGit committed:
untracked dir

MatInGit committed:
re-tracked

MatInGit committed:
sphinx doc worflow

MatInGit committed:
sphinx doc worflow

MatInGit committed:
sphinx doc worflow

MatInGit committed:
sphinx doc worflow

MatInGit committed:
sphinx doc workflow

MatInGit committed:
sphinx doc workflow

MatInGit committed:
sphinx doc workflow re-added static path

MatInGit committed:
sphinx doc workflow changed deploy script

MatInGit committed:
sphinx doc workflow changed deploy script

MatInGit committed:
sphinx doc workflow more deploy script changes

MatInGit committed:
sphinx doc workflow more deploy script changes

MatInGit committed:
sphinx doc workflow more deploy script changes

MatInGit committed:
changed makefile to make api docs

MatInGit committed:
changed makefile to make api docs

MatInGit committed:
changed makefile to make api docs

MatInGit committed:
changed makefile to make api docs

MatInGit committed:
changed the workflow to just use the built docs

MatInGit committed:
updated git ignore

MatInGit committed:
gitignore changes

MatInGit committed:
TOC change

MatInGit committed:
TOC change

MatInGit committed:
TOC change

MatInGit committed:
TOC change

MatInGit committed:
added warning

MatInGit committed:
added warning

MatInGit committed:
cool new theme

MatInGit committed:
clean docs build

MatInGit committed:
docs docs docs

MatInGit committed:
source

MatInGit committed:
docs work

MatInGit committed:
removed priority legend

MatInGit committed:
more readable roadmap

MatInGit committed:
finished readme for release

MatInGit committed:
Add simple model training notebook and update test for LocalModelGetter

MatInGit committed:
base loacl example completed

MatInGit committed:
updated local deployment instructions

MatInGit committed:
updated local deployment instructions

MatInGit committed:
Refactor model getters and add legacy support

- Updated LocalModelGetter to improve logging format.
- Refactored MLflowModelGetter to introduce a legacy class (MLflowModelGetterLegacy) and added warnings for its usage.
- Enhanced MLflowModelGetter to handle model URIs and versions more flexibly.
- Updated __init__.py to register the legacy model getter.
- Added CPU usage tracking in messaging.py for better performance monitoring.
- Created a new YAML configuration for testing the legacy model getter.
- Implemented tests for the legacy model getter to ensure functionality.
- Added a script to list registered models in MLflow for easier management.
- Cleaned up unnecessary whitespace and comments in various files.

MatInGit committed:
added lume-model examples, not working yet

MatInGit committed:
cleanup

MatInGit committed:
all tests passing

MatInGit committed:
copy-pasta fix

MatInGit committed:
Merge pull request #22 from MatInGit/20-documentation

Improved docs

MatInGit committed:
pypi publish

MatInGit committed:
added branch 18 to ci/cd for testing

MatInGit committed:
fixed publish action

MatInGit committed:
skipping test if mlflow not reachable

MatInGit committed:
fixed testing

MatInGit committed:
removed workspace file

MatInGit committed:
test change

MatInGit committed:
fixed test pypi repo

MatInGit committed:
pypi keep testing

MatInGit committed:
changed the ci/cd flow

MatInGit committed:
merged

MatInGit committed:
merge fix

MatInGit committed:
update build script

MatInGit committed:
docker ignore everything

MatInGit committed:
build prep

MatInGit committed:
build prep

MatInGit committed:
dev7

MatInGit committed:
Update Dockerfile and related scripts for poly-lithic integration and dependency adjustments

MatInGit committed:
Refactor docker-compose.yml to comment out volume mappings and command for clarity

MatInGit committed:
Update CI workflow, Docker configurations, and README for improved deployment and package installation

MatInGit committed:
Add setup cell for poly-lithic installation in Jupyter notebook and update README with correct Docker command

MatInGit committed:
Update GitHub workflows and example notebooks

- Change the branch for Sphinx Docs deployment from '18-improved-model-getting-and-examples' to 'dev'.
- Update main workflow to remove the '18-improved-model-getting-and-examples' branch.
- Modify Docker image build script comment for clarity regarding version tagging.
- Enhance simple_model_local notebook by adding output for pip install command.
- Add setup cell in simple_model_mlflow notebook for poly-lithic installation.

MatInGit committed:
Add GitHub Actions workflow for building and pushing Docker images on tag

MatInGit committed:
Refactor Docker build workflow to improve semver check and update tagging logic
